web-api/src/personalization/routes.rs

```diff
@@ -333,11 +331,9 @@ pub(crate) async fn personalize_documents_by(
     let mut documents_by_tags = all_documents
         .iter()
         .map(|document| {
-            let weight = document
-                .tags
-                .as_deref()
-                .and_then(|tag| tags.get(tag).copied())
-                .unwrap_or_default();
+            let weight = document.tags.iter().fold(0, |weights, tag| {
+                tags.get(tag).map_or(weights, |weight| weights + *weight)
+            });
```

Comment:
:art:
I think this could be also written as something like:
```suggestion
            let weight = document.tags.iter().map(|tag| tags.get(tag).unwrap_or(0)).sum();
```